server/src/main/java/io/crate/planner/operators/GroupHashAggregate.java

```diff
@@ -67,17 +67,13 @@ public class GroupHashAggregate extends ForwardingLogicalPlan {
     final List<Function> aggregates;
     final List<Symbol> groupKeys;
     private final List<Symbol> outputs;
-    private final long numExpectedRows;
 
 
-    static long approximateDistinctValues(Stats stats, List<Symbol> groupKeys) {
+    public static long approximateDistinctValues(Stats stats, List<Symbol> groupKeys) {
         long numSourceRows = stats.numDocs();
         long distinctValues = 1;
         int numKeysWithStats = 0;
         for (Symbol groupKey : groupKeys) {
-            if (Symbols.containsCorrelatedSubQuery(groupKey)) {
-                throw new UnsupportedOperationException("Cannot use correlated subquery in GROUP BY clause");
-            }
```

Comment:
Looks like we have this check only on `GroupHashAggregate` creation. Does it mean we lost that check in calls of static method, for example[ new usage](https://github.com/crate/crate/pull/14228/files#diff-0e8a207c2fafdd49468e21355bde43ef513a271b189795e8ca736e4add4bcd2dR216) in PlanStats?

Reply:
The linked call is on a `GroupHashAggregate` instance, which means the logic in the constructor must have run already.